src/main/webapp/scripts/counter.js

```diff
@@ -0,0 +1,37 @@
+let loadingCounter = (function(value = 0) {
```

Comment:
[optional] it may be more intuitive to make loadingCounter a class, but it's your call.

Reply:
In javascript it is preferred to use closures and modules over classes.

Reply:
Juan and I discussed this today as well.  If this were Java I would support making this a Class but in Javascript I agree that the use of closures over Classes is the preferred style.

Juan, regarding our discussion of using Reflection in a callback were you able to find out any more about that?  I'll approve this for now but in a future PR it might be nice to abstract the hide functionality outside of the counter module.

Reply:
Yeah, I just got out PR #62 tracing the counter methods.